subbasins.py

```diff
@@ -120,12 +124,12 @@ def delineate(input_csv: str, output_prefix: str):
         outlets = gage_basins_dict[megabasin]
 
         if VERBOSE: print('Reading geodata for unit catchments in megabasin %s' % megabasin)
-        catchments_gdf = load_gdf("catchments", megabasin, True)
+        catchments_gdf = load_gdf("catchments", megabasin, True, bounds)
 
         # The _network_ data is in the RIVERS file rather than the CATCHMENTS file
         # (this is just how the MERIT-Basins authors did it)
         if VERBOSE: print('Reading geodata for rivers in megabasin %s' % megabasin)
-        rivers_gdf = load_gdf("rivers", megabasin, True)
+        rivers_gdf = load_gdf("rivers", megabasin, True, bounds)
```

Comment:
we timed this before (reading from the locally downloaded files) and after (reading from the global remote file) and saw an increase from .04 to 4 seconds

this is offset by the code to lookup the pfastater 2 basin, which takes 12 seconds and that we'll be able to 🔪 b/c of these changes 

we will also be looking into adding a public domain in r2 w/ caching to speed duplicate requests up